compiler/damlc/tests/daml-test-files/SymbolKindWarnings.daml

```diff
@@ -0,0 +1,26 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
```

Comment:
We should probably also add a couple of tests to verify that indirectly introducing a `Symbol` constrain via `HasField` does not give false warnings. I'm totally aware that this implementation has no chance of producing such bogus warnings. However, if we ever need to move the check past the type checker, for whatever reasons, this would be easy to break.

Reply:
Makes sense, I'll add a couple of HasField instances & constraints for good measure.